.github/workflows/faucet.yml

```diff
@@ -0,0 +1,79 @@
+name: Faucet
+on:
+  repository_dispatch:
+    types: [sync-contracts]
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+  
+jobs:
+  update_list:
+    name: Fetch & Update Faucet Tokens
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          cache: 'yarn'
+      - name: Run Script
+        run: |
+          yarn install --frozen-lockfile
+          node src/fetchFaucetTokens.js
+          npm run prettier
+        shell: bash
+
+      - name: Commit Update
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
+          if [ -n "$(git status --porcelain)" ]; then
+            git pull
+            git checkout -f "Automated-FaucetToken-Fetch-Update"
+            ./run.sh
+            npm run prettier
+            git add .
+            git commit -m "Updated EVM TokenList" || echo  "Branch is up to date"
+            git push -u origin "Automated-FaucetToken-Fetch-Update"  
+            gh pr create -B main -H Automated-FaucetToken-Fetch-Update --title 'EVM TokenList Updated' --body  'PR created by Github Actions. Review PR and add changeset to release' || echo "PR already exists"
+          else 
+            echo "Repo is clean"
+          fi
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  release:
+    if: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
We don't need this check, since this workflow will be only run on the default branch which is main.